beancount_n26/__init__.py

```diff
@@ -256,53 +268,28 @@ def extract(self, filepath: str, existing: data.Entries = None) -> data.Entries:
         s_exchange_rate = self._translate("exchange_rate")
 
         with open(filepath, encoding=self.file_encoding) as fd:
+            reader_with_lines = LastLineReader(fd)
```

Comment:
I think we could avoid this `LastLineReader` class if we collect all the lines in a list first and use that to set `meta["__source__"]`.

```python
lines = [line.strip() for line in fd.readlines()]
reader = csv.DictReader(lines, ...)

for index, line in enumerate(reader):
    meta["__source__"] = lines[index]
```

Reply:
I've added it specifically to avoid reading all lines to memory.